### PR TITLE
Add error detail popup

### PR DESCRIPTION
**Pull request by bhavenst**
Adds a popup with error, error details and correlation ID.  Not sure of the best way to do the CSS stuff, so let me know.

### CODE REVIEW

**Comment by bhavenst** [comment]
image

File: ui/src/app/ErrorInfo/ErrorInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { Popover, Icon, Tooltip } from '@patternfly/react-core';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/esm/icons/exclamation-circle-icon';
+import { DeploymentStepStatusData } from '@app/apis/types';
+
+interface IErrorInfoProps {
+  stepStatusData: DeploymentStepStatusData;
+}
+
+export const ErrorInfoPopover = ({ stepStatusData }: IErrorInfoProps) => {
+  const borders = {
```
Comment (tznamena) [inline ui/src/app/ErrorInfo/ErrorInfo.tsx:11]: Let's move this out of the component and into a css file. Then let's add class names.

File: ui/src/app/ErrorInfo/ErrorInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    border: '1px solid rgb(0,0,0)',
+  };
+
+  let data = (
```
Comment (tznamena) [inline ui/src/app/ErrorInfo/ErrorInfo.tsx:15]: This should be `const`. 
Also, how about doing this with patternfly's table?
https://www.patternfly.org/v4/components/table#composable-basic
Or we can do it in another PR.

**Review by tznamena — CHANGES_REQUESTED** [review]

**Comment by bhavenst** [comment]
Updated image after switching to the patternfly table component..  
image

File: ui/src/app/ErrorInfo/ErrorInfo.tsx [hunk 3]
```diff
@@ -1,38 +1,41 @@
...
   stepStatusData: DeploymentStepStatusData;
 }
 
 export const ErrorInfoPopover = ({ stepStatusData }: IErrorInfoProps) => {
-
   const data = (
     <>
-    <TableComposable>
-        <Tr>
+      <TableComposable>
```
Comment (tznamena) [inline ui/src/app/ErrorInfo/ErrorInfo.tsx:16]: Since this has one top level element we don't need to wrap it in `<> </>`.

File: ui/src/app/ErrorInfo/ErrorInfo.tsx [hunk 4]
```diff
@@ -1,38 +1,41 @@
...
-    </TableComposable>
+          </Tr>
+        </Tbody>
+      </TableComposable>
     </>
   );
   return (
-    <Tooltip content={<div>Click for error info</div>}>
-      <Popover minWidth="40rem" bodyContent={<div>{data}</div>}>
+    <Tooltip removeFindDomNode={true} content={<div>Click for error info</div>}>
```
Comment (tznamena) [inline ui/src/app/ErrorInfo/ErrorInfo.tsx:37]: I never see this text when hovering over the icon. Only time it shows up is when I comment out the `Popover` and leave the `Icon` in.

File: ui/src/app/ErrorInfo/ErrorInfo.tsx [hunk 5]
```diff
@@ -1,38 +1,41 @@
...
+      <Popover removeFindDomNode={true} minWidth="40rem" bodyContent={<div>{data}</div>}>
```
Comment (tznamena) [inline ui/src/app/ErrorInfo/ErrorInfo.tsx:38]: No need to wrap the `data` in a `<div> </div>` it already has only one top level element.

File: ui/src/app/ErrorInfo/ErrorInfo.tsx (hunk 3 above)
Comment (bhavenst) [reply ui/src/app/ErrorInfo/ErrorInfo.tsx:16]: OK!

File: ui/src/app/ErrorInfo/ErrorInfo.tsx (hunk 5 above)
Comment (bhavenst) [reply ui/src/app/ErrorInfo/ErrorInfo.tsx]: OK!

File: ui/src/app/ErrorInfo/ErrorInfo.tsx (hunk 4 above)
Comment (bhavenst) [reply ui/src/app/ErrorInfo/ErrorInfo.tsx]: This happens due to the removeFindDomNode tag.  So I removed that again and disabled strict mode to get rid of the warning.

**Review by benjdudas — APPROVED** [review]
Looks good to me

**Review by tznamena — APPROVED** [review]
